# HG changeset patch
# User Rik <rik@octave.org>
# Date 1382041178 25200
#      Thu Oct 17 13:19:38 2013 -0700
# Node ID c14e5af64de4d830fbd9f2842bf312dfb10f8043
# Parent  c060ad097056ff3037191412486c7dc1d7da3c6b
legend.m: Fix invalid legend handle with plotyy (bug #35321).

* scripts/plot/appearance/legend.m: Correctly get axes children when
there are more than one axes (as is the case for plotyy).

diff --git a/scripts/plot/appearance/legend.m b/scripts/plot/appearance/legend.m
--- a/scripts/plot/appearance/legend.m
+++ b/scripts/plot/appearance/legend.m
@@ -155,17 +155,17 @@ function [hleg, hleg_obj, hplot, labels]
     kids = flipud (varargin{1}(:));
     varargin(1) = [];
   else
     kids = ca;
     kids(strcmp (get (ca, "tag"), "legend")) = [];
     if (isscalar (kids))
       kids = get (kids, "children")(:);
     else
-      kids = flipud ([get(kids, "children"){:}](:));
+      kids = flipud (vertcat (get (kids, "children"){:}));
     endif
   endif
   nargs = numel (varargin);
   nkids = numel (kids);
 
   orientation = "default";
   location = "default";
   show = "create";
